rocket-lazy-load.php

```diff
@@ -484,41 +525,46 @@ function rocket_lazyload_iframes( $html ) {
 				continue;
 			}
 
+			$query = wp_parse_url( $iframe[1], PHP_URL_QUERY );
+
 			/**
 			 * Filter the LazyLoad HTML output on Youtube iframes
 			 *
 			 * @since 1.4.3
 			 *
 			 * @param array $html Output that will be printed.
 			 */
-			$youtube_lazyload = apply_filters( 'rocket_lazyload_youtube_html', '<div class="rll-youtube-player" data-id="' . $youtube_id . '"></div>');
+			$youtube_lazyload  = apply_filters( 'rocket_lazyload_youtube_html', '<div class="rll-youtube-player" data-id="' . esc_attr( $youtube_id ) . '" data-query="' . esc_attr( $query ) . '"></div>' );
 			$youtube_lazyload .= '<noscript>' . $iframe[0] . '</noscript>';
 
 			$html = str_replace( $iframe[0], $youtube_lazyload, $html );
-		} else {
-			/**
-			 * Filter the LazyLoad placeholder on src attribute
-	    	 *
-	    	 * @since 1.1
-	    	 *
-	    	 * @param string $placeholder placeholder that will be printed.
-	    	 */
-			$placeholder = apply_filters( 'rocket_lazyload_placeholder', 'about:blank' );
-			
-			$iframe_noscript = '<noscript>' . $iframe[0] . '</noscript>';
 
-			/**
-			 * Filter the LazyLoad HTML output on iframes
-			 *
-			 * @since 1.1
-			 *
-			 * @param array $html Output that will be printed.
-			 */
-			$iframe_lazyload = apply_filters( 'rocket_lazyload_iframe_html', str_replace( $iframe[1], $placeholder . '" data-rocket-lazyload="fitvidscompatible" data-lazy-src="' . $iframe[1], $iframe[0] ) );
-			$iframe_lazyload .= $iframe_noscript;
-			
-			$html = str_replace( $iframe[0], $iframe_lazyload, $html );
+			continue;
 		}
+
+		/**
+		 * Filter the LazyLoad placeholder on src attribute
+		 *
+		 * @since 1.1
+		 *
+		 * @param string $placeholder placeholder that will be printed.
+		 */
+		$placeholder = apply_filters( 'rocket_lazyload_placeholder', 'about:blank' );
+
+		$iframe_noscript = '<noscript>' . $iframe[0] . '</noscript>';
+
+		$iframe_lazyload = str_replace( $iframe[1], $placeholder, $iframe[0] );
+		/**
+		 * Filter the LazyLoad HTML output on iframes
+		 *
+		 * @since 1.1
+		 *
+		 * @param array $html Output that will be printed.
+		 */
+		$iframe_lazyload  = apply_filters( 'rocket_lazyload_iframe_html', str_replace( $iframe[2], $iframe[2] . ' data-rocket-lazyload="fitvidscompatible" data-lazy-src="' . $iframe[1] . '"', $iframe[0] ) );
```

Comment:
This `$iframe[2]` thing has many ways to fail.
Not set at all:
```html
<iframe src="http://example.com"></iframe>
```
A space character (my favorite, because the attributes will be added everywhere a space character is present :D):
```html
<iframe title="Oh, hey!" src="http://example.com" ></iframe>
```

So, here is what I would do instead.
First of all, improve the first regex ([tests](https://regex101.com/r/moDSAK/3)):
```php
preg_match_all( '/<iframe.*\ssrc=["|\'](.+)["|\'].*(>\s*<\/iframe>)/iU', $html, $matches, PREG_SET_ORDER );
```
Then:
```php
$iframe_lazyload  = str_replace( $iframe[2], ' data-rocket-lazyload="fitvidscompatible" data-lazy-src="' . esc_url( $iframe[1] ) . '"' . $iframe[2], $iframe[0] );
/** DocBlock */
$iframe_lazyload  = apply_filters( 'rocket_lazyload_iframe_html', $iframe_lazyload );
$iframe_lazyload .= $iframe_noscript;
```